services/configuration/src/schema.rs

```diff
@@ -0,0 +1,272 @@
+        MaybeVote(None)
+    }
+
+    /// Creates a `Some` variant.
+    pub fn some(vote: Vote) -> Self {
+        MaybeVote(Some(vote))
+    }
+}
+
+impl From<MaybeVote> for Option<Vote> {
+    fn from(value: MaybeVote) -> Option<Vote> {
+        value.0
+    }
+}
+
+impl ::std::ops::Deref for MaybeVote {
+    type Target = Option<Vote>;
+
+    fn deref(&self) -> &Option<Vote> {
+        &self.0
+    }
+}
+
+impl CryptoHash for MaybeVote {
+    fn hash(&self) -> Hash {
+        match self.0 {
+            Some(ref vote) => vote.hash(),
+            None => crypto::hash(&NO_VOTE_BYTES),
+        }
+    }
+}
+
+impl StorageValue for MaybeVote {
+    fn into_bytes(self) -> Vec<u8> {
+        match self.0 {
+            Some(vote) => vote.into_bytes(),
+            None => NO_VOTE_BYTES.clone(),
+        }
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        if NO_VOTE_BYTES.as_slice().eq(bytes.as_ref()) {
+            MaybeVote::none()
+        } else {
+            MaybeVote::some(Vote::from_bytes(bytes))
+        }
+    }
+}
+
+/// Database schema used by the configuration service.
+#[derive(Debug)]
+pub struct Schema<T> {
+    view: T,
+}
+
+impl<T> Schema<T>
+where
+    T: AsRef<Snapshot>,
+{
+    /// Creates a new schema.
+    pub fn new(snapshot: T) -> Schema<T> {
+        Schema { view: snapshot }
+    }
+
+    /// Returns propose information indexed by the hash of the configuration corresponding
+    /// to a proposal.
+    ///
+    /// Consult [the crate-level docs](index.html) for details how hashes of the configuration
+    /// are calculated.
+    pub fn propose_data_by_config_hash(&self) -> ProofMapIndex<&Snapshot, Hash, ProposeData> {
+        ProofMapIndex::new(PROPOSES, self.view.as_ref())
+    }
+
+    /// Returns a table of hashes of proposed configurations in the commit order.
+    pub fn config_hash_by_ordinal(&self) -> ProofListIndex<&Snapshot, Hash> {
+        ProofListIndex::new(PROPOSE_HASHES, self.view.as_ref())
+    }
+
+    /// Returns a table of votes of validators for a particular proposal, referenced
+    /// by its configuration hash.
+    pub fn votes_by_config_hash(&self, config_hash: &Hash) -> ProofListIndex<&Snapshot, MaybeVote> {
+        ProofListIndex::with_prefix(VOTES, gen_prefix(config_hash), self.view.as_ref())
+    }
+
+    /// Returns a `Propose` transaction with a particular configuration hash.
+    pub fn propose(&self, cfg_hash: &Hash) -> Option<Propose> {
+        self.propose_data_by_config_hash().get(cfg_hash).map(
+            |propose_data| propose_data.tx_propose(),
+        )
+    }
+
+    /// Returns a list of votes for the proposal corresponding to the given configuration hash.
+    #[cfg_attr(feature = "cargo-clippy", allow(let_and_return))]
+    pub fn votes(&self, cfg_hash: &Hash) -> Vec<Option<Vote>> {
+        let votes_by_config_hash = self.votes_by_config_hash(cfg_hash);
+        let votes = votes_by_config_hash.iter().map(MaybeVote::into).collect();
+        votes
```

Comment:
`votes` variable does not seem to be necessary. Given that the clippy warning is silence it seems that this is intentional, but it doesn't seem to make code more readable?